pyiron_contrib/atomistics/atomicrex/structure_list.py

```diff
@@ -299,6 +303,11 @@ def get_vector_property(self, prop, identifier, final=True):
         else:
             return self.fit_properties[prop]._per_element_arrays["target_val"][slc]
 
+    def _sync(self):
```

Comment:
This is done to prevent the dangerous situation that num_elements and num_chunks are different throughout the several Flattened storages that I use to store properties. In most of the code I just use the indices stored in self._structures directly
and never actually change the num_elements or num_chunks, but here  I need to sync them to be able to use the extend function on the fit property FlattenedStorages.

TBH I think my current implementation offers a lot of potentially dangerous ways to mess with the code due to the history of this class and this could help to reduce these risks. Probably I should do a complete rework, maybe by modifying the strings directly and storing everything in 1 FlattenedStorage, or by defining everything as properties (also open for other suggestions) but that is for another PR

Reply:
Yeah, it can stay that way, I just want to have a feeling of what is needed in the class.

I thought about the concept of linking multiple storages together so that their arrays stay in sync, but felt that having one big storage is simpler.  It should also be fairly simple to break the big container into smaller ones again with `.split` in a property somewhere.